### ISSUE TITLE
R&D - исследования и разработка

### ISSUE THREAD

**Issue by TelmerTemp** [klensy/wt-tools#26]
1. с сайта нельзя достать описание старых реплеев или найти из по дате
2. источники реплеев только улитки(серверные реплеи) или игроки/комьюнити. Но так как новый клиент игры не открывает старые реплеи - ИМХО мало кто их хранит
3. battle_id(в девичестве data-replay) - оочень похож на unix-time c 10 наносекундным разрешением
поэтому поэтому можно unix-time домножить на 76416.016 и поискать рядом
4. из battle id легко найти ссылку - преобразовать в шетнадцатеричный формат и добавить ведущий ноль.

**Comment by TelmerTemp** [comment]
есть ли какое-небудь ограничение на глубину вложенности блоков в blk формате?

Пока получается г-код с обьектом, в котором хранится обьект этого же класса, к котором может храниться /и так до бесконечности/

**Comment by diwersan7** [comment]
Не думаю
Лимит есть только у декодера( например у встроенного в PHP json_decode максимально допустимая глубина 512)

**Comment by TelmerTemp** [comment]
что-то не выходит у меня каменный цветок....

зато могу подсказать по получению списка реплеев

**Comment by klensy** [comment]
Нет, ограничения вложенности я не делал, т.к. не встречал больше 5 уровня глубины.

**Comment by TelmerTemp** [comment]
пытаюсь сделать на java - получается такой махровый г-код :)
кроме моих знаний дополнительной ограничение - строгая типизация джавы и отсутствие обобщений(generics) в рантайме.
ну и не до конца понял структуру blk

**Comment by diwersan7** [comment]
Что за подсказка по списку реплеев?

**Comment by TelmerTemp** [comment]
серверные реплеи не хранят таблицу фрагов и очков. Из обычных они извлекается с помощью wt-tools. а что можно извлечь из серверных - я пока не знаю.

-на первое время авторизацию можно делать кукисами
-не обязательно скачивать собирать базу на все реплеи - можно по сгененированному числу попытаться попасть в какой-небудь.

**Comment by diwersan7** [comment]
Они как-то соединяются?

**Comment by TelmerTemp** [comment]
если в папку с клиентом игры загрузить обычный реплей - будет видна статистика. если серверный - то нет
на глаз отдельные места похожи на blk(а есть еще BBF куски, которые распаковываются этой версией и сжатые deflate блоки)

нужна помощь в написании программы, которая будет разбирать файлы

**Comment by diwersan7** [comment]
для этого нужен @klensy , если у него есть время и желание.

**Comment by TelmerTemp** [comment]
X3E7MO@jabber.ru мой джаббер

**Comment by TelmerTemp** [comment]
@klensy тоже вроде пишет на питоне. А в языках с нестрогой типизацией или ее отсутствием несколько проще :)

**Comment by Ferrocen** [comment]
Как на счет исследования WT для хаккинга геймплея?

**Comment by diwersan7** [comment]
@Ferrocen Если тебе нужны читы - иди на соответсвующий форум и покупай.

**Comment by klensy** [comment]
@Ferrocen 
Данные инструменты для довольно безобидной цели - разбора реплеев, настроек клиента и распаковки ресурсов. Твои вопросы уходят в довольно темную область, которой не место *тут*. Возможно, в каком-то другом месте это и допустимо.

**Comment by TelmerTemp** [comment]
@Ferrocen
Если исследования помогут пролить свет на то, как хранятся данные в реплее и что они означают - мы будем признательны

**Comment by penlu** [comment]
I am interested in parsing the wrpl file format, especially to help this project:
https://github.com/diVineProportion/ThunderTac

I would like to ask what secrets this R&D may have revealed. I would like to help if I can.

Thanks!

**Comment by klensy** [comment]
@penlu I really didn't checked anything from that old time.
There few projects, that works with localhost browser tracker.

**Comment by penlu** [comment]
I understand, thank you!

I will continue to try to understand the contents of WRPLU files, as you have named them. I would appreciate any advice you could provide!
